crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -1181,18 +1223,26 @@ impl CheckpointBuilder {
             {
                 let (transaction, size) = transaction_and_size
                     .unwrap_or_else(|| panic!("Could not find executed transaction {:?}", effects));
-                // ConsensusCommitPrologue and AuthenticatorStateUpdate are guaranteed to be
-                // processed before we reach here
-                if !matches!(
-                    transaction.inner().transaction_data().kind(),
+                match transaction.inner().transaction_data().kind() {
                     TransactionKind::ConsensusCommitPrologue(_)
-                        | TransactionKind::ConsensusCommitPrologueV2(_)
-                        | TransactionKind::AuthenticatorStateUpdate(_)
-                        | TransactionKind::RandomnessStateUpdate(_)
-                ) {
-                    transaction_keys.push(SequencedConsensusTransactionKey::External(
-                        ConsensusTransactionKey::Certificate(*effects.transaction_digest()),
-                    ));
+                    | TransactionKind::ConsensusCommitPrologueV2(_)
+                    | TransactionKind::AuthenticatorStateUpdate(_) => {
+                        // ConsensusCommitPrologue and AuthenticatorStateUpdate are guaranteed to be
+                        // processed before we reach here.
+                    }
+                    TransactionKind::RandomnessStateUpdate(rsu) => {
+                        // RandomnessStateUpdate does not come via consensus, so no need to include
+                        // it in the call to `consensus_messages_processed_notify`.
```

Comment:
this comment seems confusing / out of date

Reply:
confusing maybe, but not out of date, I just updated it :) it's perhaps overly verbose anyway so I just deleted it